force-app/main/default/classes/newTaskTraining.cls

```diff
@@ -0,0 +1,18 @@
+public class newTaskTraining {
+    public static void createTask() {
+        Date today = Date.today();
+        List<Training__c> endedTrains = [SELECT Name, End_Date__c, Training_Course__r.OwnerId
+                                        FROM Training__c
+                                        WHERE DAY_ONLY(End_Date__c) = :today];
+        //System.debug(endedTrains);
+        List<Task> taski = new List<Task>();
+        for(Training__c train :endedTrains) {
+            Task task = new Task(Subject = 'Fill GPA (marks) and mark all participants as Participated.',
```

Comment:
I suggest creating a new method like "createTask()" that will accept all the parameters and return a new task record. It will make your method shorter and code more readable.